pkg/detectors/easyinsight/easyinsight.go

```diff
@@ -60,27 +64,37 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			}
 
 			if verify {
-				data := fmt.Sprintf("%s:%s", resIdMatch, resMatch)
-				sEnc := b64.StdEncoding.EncodeToString([]byte(data))
+				auth := fmt.Sprintf("%s:%s", resIdMatch, resMatch)
+				sEnc := b64.StdEncoding.EncodeToString([]byte(auth))
+
 				req, err := http.NewRequestWithContext(ctx, "GET", "https://www.easy-insight.com/app/api/users.json", nil)
 				if err != nil {
 					continue
 				}
+
+				// add required headers to the request
 				req.Header.Add("Content-Type", "application/json")
 				req.Header.Add("Accept", "application/json")
 				req.Header.Add("Authorization", fmt.Sprintf("Basic %s", sEnc))
+
 				res, err := client.Do(req)
 				if err == nil {
-					defer res.Body.Close()
+					// discard the body content and close it at the end of each iteration.
+					_, _ = io.Copy(io.Discard, res.Body)
+					_ = res.Body.Close()
+
 					if res.StatusCode >= 200 && res.StatusCode < 300 {
```

Comment:
We should pick specific status codes that are considered valid. Permissive ranges like this have been a source of many false positives.

Reply:
Documentation actually does not provide any information about which status code the API will return for different scenario.

Reply:
@kashifkhan0771 Here is the [documentation](https://www.easy-insight.com/api/responses.html) about supported HTTP Status codes.

Reply:
Thanks @abmussani!